### PR TITLE
RA-1875: EMPT66 Fixed XSS Vulnerability in Schedule Form

### PR DESCRIPTION
**Pull request by The-Lady**
### Description of What I Changed

I added encoding to SchedulerForm java file to prevent XSS vulnerability in Task Name.

### Issue I Worked On

An iframe/script could be injected in Task Name while adding a new task in Scheduler. I encoded the name of the task in the controller file to mitigate this.

Steps to reproduce the vulnerability:

1. Launch the OpenMRS application.
2. Login with username "Admin" and password "Admin123" with location as Inpatient Ward.
3. Click on the “System Administration” option and then on "Advanced Administration".
4. Click on “Manage Scheduler”.
5. Click on “Add Tasks”.
6. Enter '<iframe src="https://www.csc.ncsu.edu/">' as Name and “org.openmrs.scheduler.tasks.SendEmailTask” and Schedulable Class.
7. Click on “Save”.

Output: An iframe will be displayed.

### Link to ticket
[RA-1875](https://issues.openmrs.org/browse/RA-1875)

@isears

### CODE REVIEW

File: omod/src/main/java/org/openmrs/web/dwr/DWRConceptService.java [hunk 1]
```diff
@@ -222,10 +222,8 @@
 			
 			if (searchResults.size() < 1) {
 				String htmlSafePhrase = "<b>" + Encode.forHtml(phrase) + "</b>";
-				objectList.add(Context.getMessageSourceService()
-				        .getMessage("general.noMatchesFoundInLocale",
-				            new Object[] { htmlSafePhrase, OpenmrsUtil.join(searchLocales, ", ") },
-				            Context.getLocale()));
+				objectList.add(Context.getMessageSourceService().getMessage("general.noMatchesFoundInLocale",
```
Comment (HerbertYiga) [inline omod/src/main/java/org/openmrs/web/dwr/DWRConceptService.java:225]: @The-Lady  did you make any code changes around this place?

**Review by isears — CHANGES_REQUESTED** [review]
@The-Lady this looks good but it seems like the changes to `DWRConceptService.java` aren't necessary (as @HerbertYiga pointed out).

If that's the case I'll go ahead and merge this in after the `DWRConceptService.java` changes are reset.

File: omod/src/main/java/org/openmrs/web/dwr/DWRConceptService.java (hunk 1 above)
Comment (The-Lady) [reply omod/src/main/java/org/openmrs/web/dwr/DWRConceptService.java]: @HerbertYiga I am sorry, I did not notice the unintentional change in indentation. I have reverted the change and committed to the branch.

**Comment by The-Lady** [comment]
@isears I have reset the changes in `DWRConceptService.java`

**Comment by Parth59** [comment]
Hi @The-Lady .  Can we try to import the org.openmrs.web.WebUtil package to utilize WebUtil.escapeHTML() function () ? @isears Will it be a better idea ?

**Review by isears — CHANGES_REQUESTED** [review]
There's still changes in DWRConceptService.java as part of this PR. Just revert the file to the original so that we can keep the history clean. Something like this should do the trick: 
```
git checkout origin/master omod/src/main/java/org/openmrs/web/dwr/DWRConceptService.java 
```
> Hi @The-Lady . Can we try to import the org.openmrs.web.WebUtil package to utilize WebUtil.escapeHTML() function () ? @isears Will it be a better idea ?

Yes, in general the webutil escaping functions (which are backed by OWASP) are more robust than apache string escape utils. If it's possible, let's use webutil, but if it creates issues don't worry to much about it.

**Comment by The-Lady** [comment]
@Parth59 Thank you for the suggestion

@isears I have reverted the DWRConcept file to original and switched to WebUtil for encoding as suggested.

**Comment by isears** [comment]
@The-Lady thanks for switching to webUtil, that looks good.

It looks like `DWRConceptService.java` is still showing up as a change in the git diff of this PR. Were you able to run the git command I posted above? The end result should be that only `SchedulerFormController.java` shows up in the "Files changed" tab.

The reason it's so important to not include modification to files that don't have to be modified as part of a patch is so that we can more easily trace the origins of bugs that get introduced into the codebase.

**Comment by The-Lady** [comment]
@isears The indentation kept getting changed whenever I run the application on localhost. I have reverted it again.

**Review by isears — APPROVED** [review]
Great, thanks @The-Lady !

**Pull request merged**